core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -938,9 +945,13 @@ def _generate_fold_configs(
                     folds_to_fit=folds_to_fit,
                     folds_finished=fold - fold_start,
                     folds_left=fold_end - fold,
+                    random_seed=random_seed_value
                 )
 
                 fold_fit_args_list.append(fold_ctx)
+                if vary_seed_across_folds:
+                    random_seed_value += 1
```

Comment:
This is technically incorrect in the scenario where `n_repeat_start` and `fold_in_set_start` don't equal 0, such as when fitting an extra repeat of the bag in a future fit call.

The way to do it correctly would be to ensure the Nth child model gets a random_seed_value of `n-1`. Code will be a bit more complicated, but pretty sure there is a smart way to do it, as I've already had to deal with this same problem in the past somewhere in the bagged ensemble code.